### PR TITLE
[EOS-18495] Cleanup salt packages and dir using salt states

### PR DESCRIPTION
**Pull request by mazinamdar**
Signed-off-by: mazinamdar <mazhar.inamdar@seagate.com>

### CODE REVIEW

**Review by andkononykhin2 — APPROVED** [review]

File: srv/components/provisioner/salt/teardown/remove_dir.sls [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
```
Comment (ypise) [inline srv/components/provisioner/salt/teardown/remove_dir.sls:16]: Also cleanup in respective teardown directories:
`/var/cache/salt`

File: srv/components/provisioner/salt/teardown/stop.sls [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
```
Comment (ypise) [inline srv/components/provisioner/salt/teardown/stop.sls:16]: Change the directory of file:
`srv/components/provisioner/salt/teardown/stop.sls`

**Review by ypise — CHANGES_REQUESTED** [review]
Move the configs:
`srv/components/provisioner/salt_master/config.sls` & `srv/components/provisioner/salt_minion/config.sls` to
`srv/components/provisioner/salt/config/master.sls` & `srv/components/provisioner/salt/config/minion.sls`

Also the files go to `srv/components/provisioner/salt/files/`

We'll have to ultimately move salt under 3rd party SW directory, i.e. under `srv/components/misc_pkgs`.

File: srv/components/provisioner/salt/teardown/remove_dir.sls (hunk 1 above)
Comment (mazinamdar) [reply srv/components/provisioner/salt/teardown/remove_dir.sls:16]: comment is addressed

File: srv/components/provisioner/salt/teardown/stop.sls (hunk 2 above)
Comment (mazinamdar) [reply srv/components/provisioner/salt/teardown/stop.sls]: comment is addressed

**Review by ypise — APPROVED** [review]

**Pull request merged**